models/modules/palette_denoise_fn.py

```diff
@@ -101,7 +101,12 @@ def forward(self, input, embed_noise_level, cls, mask, ref):
         if "mask" in self.conditioning:
             input = torch.cat([input, mask_embed], dim=1)
 
-        return self.model(input, embedding)
+        if "Ref" in type(self.model).__name__:
```

Comment:
Is there really no other solution to do this? a flag when calling the constructor, a property or function of the model `is_ref` (which can be optional), ...

 You could use `inspect` but I'm not sure it's needed here 
 https://stackoverflow.com/questions/847936/how-can-i-find-the-number-of-arguments-of-a-python-function